### PR TITLE
[Feature/346567: Overseas reprocessor details]

### PR DESCRIPTION
**Pull request by shehzad-ismail8**

### CODE REVIEW

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        /// <param name="overseasSiteId">Overseas site ID</param>
+        /// <returns>The view result</returns>
+        [HttpGet("ReprocessorDetails")]
+        public async Task<IActionResult> ReprocessorDetails(
+            Guid? id,
+            Guid? overseasSiteId)
+        {
+            _backPageViewModel.Url = _urlHelper.ActionLink("AddOverseasReprocessingSites", "OverseasSite");
+
+            if (id == null)
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs]: Why are you not checking for overseas id?

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        /// <param name="id">This is the ID of the accreditation</param>
+        /// <param name="overseasSiteId">This is the overseas reprocessing site ID</param>
+        /// <returns>Returns the view model asynchronously</returns>
+        public async Task<ReprocessorDetailsViewModel> GetReprocessorDetailsViewModel(
+            Guid id,
+            Guid overseasSiteId)
+        {
+            var reprocessorDetailsDto = await _httpOverseasSiteService.GetReprocessorDetails(id, overseasSiteId);
+
+            var listOfCountries = await _httpCountryService.GetCountryList();
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs]: This operation should be done in the facade layer: it's what it's for: making multiple call to different end points, and combing the results into one result for the client (Portal)

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs (hunk 1 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs]: Amended

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs (hunk 2 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs]: Amended

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/HttpCountryService.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+﻿namespace EPR.Accreditation.Portal.RESTservices
+{
+    using EPR.Accreditation.Portal.Common.RESTservices;
+    using EPR.Accreditation.Portal.DTOs.Country;
+    using EPR.Accreditation.Portal.RESTservices.Interfaces;
+
+    /// <summary>
+    /// Service for the Http Country Service
+    /// </summary>
+    public class HttpCountryService : BaseHttpService, IHttpCountryService
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/HttpCountryService.cs]: Is this one needed any longer?

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/HttpOverseasSiteService.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+﻿namespace EPR.Accreditation.Portal.RESTservices
+{
+    using EPR.Accreditation.Portal.Common.RESTservices;
+    using EPR.Accreditation.Portal.DTOs.OverseasSite;
+    using EPR.Accreditation.Portal.RESTservices.Interfaces;
+
+    /// <summary>
+    /// Class for the Http overseas site service
+    /// </summary>
+    public class HttpOverseasSiteService : BaseHttpService, IHttpOverseasSiteService
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/HttpOverseasSiteService.cs:10]: As above

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/Interfaces/IHttpCountryService.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿namespace EPR.Accreditation.Portal.RESTservices.Interfaces
+{
+    using EPR.Accreditation.Portal.DTOs.Country;
+
+    /// <summary>
+    /// Interface for the Http country service
+    /// </summary>
+    public interface IHttpCountryService
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/Interfaces/IHttpCountryService.cs]: As above

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/Interfaces/IHttpOverseasSiteService.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿namespace EPR.Accreditation.Portal.RESTservices.Interfaces
+{
+    using EPR.Accreditation.Portal.DTOs.OverseasSite;
+
+    /// <summary>
+    /// Interface for the Http overseas site service
+    /// </summary>
+    public interface IHttpOverseasSiteService
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/RESTservices/Interfaces/IHttpOverseasSiteService.cs:8]: ....

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        /// <param name="id">This is the ID of the accreditation</param>
+        /// <param name="overseasSiteId">This is the overseas reprocessing site ID</param>
+        /// <returns>Returns the view model asynchronously</returns>
+        public async Task<ReprocessorDetailsViewModel> GetReprocessorDetailsViewModel(
+            Guid id,
+            Guid overseasSiteId)
+        {
+            var reprocessorDetailsDto = await _httpOverseasSiteService.GetReprocessorDetails(id, overseasSiteId);
+
+            var countries = new List<SelectListItem> { new() { Value = string.Empty, Text = ReprocessorDetailsResources.DefaultOption } };
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs]: New lines please

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+            foreach (var country in reprocessorDetailsDto.CountryList)
+            {
+                countries.Add(new() { Value = country.CountryId.ToString(), Text = country.Name });
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/OverseasSiteService.cs]: New lines

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**